mcs_benchmark_data/pipelines/physical_iqa/physical_iqa_benchmark_transformer.py

```diff
@@ -52,39 +63,24 @@ def _transform_benchmark_sample(
 
             sample = json.loads(line)
 
-            benchmark_sample_uri = URIRef(f"{dataset_uri}:sample:{sample['id']}")
-
             correct_choice = None
 
             if dataset_type != "test":
-                correct_choice = int(all_labels[i]) + 1
+                correct_choice = URIRef(
+                    f"{dataset_uri}:sample:{sample['id']}:correct_choice:{int(all_labels[i]) + 1}"
+                )
 
-            yield from _Benchmark_Transformer._prepare_sample(
-                self,
-                dataset_type=dataset_type,
+            goal = sample["goal"]
+            hypotheses = [sample["sol1"], sample["sol2"]]
+
+            yield from self._yield_goal_models(
                 dataset_uri=dataset_uri,
-                sample=sample,
+                sample_id=sample["id"],
+                concepts=None,
+                context=None,
                 correct_choice=correct_choice,
-            )
-
-            yield BenchmarkGoal(
-                uri=URIRef(f"{benchmark_sample_uri}:goal"),
-                benchmark_sample_uri=benchmark_sample_uri,
-                text=sample["goal"],
-            )
-
-            yield BenchmarkHypothesis(
-                uri=URIRef(f"{benchmark_sample_uri}:hypothesis:1"),
-                benchmark_sample_uri=benchmark_sample_uri,
-                position=1,
-                text=sample["sol1"],
-            )
-
-            yield BenchmarkHypothesis(
-                uri=URIRef(f"{benchmark_sample_uri}:hypothesis:2"),
-                benchmark_sample_uri=benchmark_sample_uri,
-                position=1,
-                text=sample["sol2"],
+                goal=goal,
+                hypotheses=hypotheses,
             )
 
             i += 1
```

Comment:
Definitely use enumerate instead of this, which is a very fragile pattern. If someone else adds a continue then this code breaks.